File: leetcode/go/1822.sign-of-the-product-of-an-array.go
func arraySign(nums []int) int {
    count := 0
    for _, n := range nums {
        if n == 0 {return 0}
        if n < 0 {count++}
    }
    if count & 1 == 0 {return 1} else {return -1}
}

File: leetcode/rust-leetcode/src/1822.sign-of-the-product-of-an-array.rs
impl Solution {
    pub fn array_sign(nums: Vec<i32>) -> i32 {
        let mut sign = 1;
        for &n in nums.iter() {
            if n == 0 {return 0;}
            if n < 0 {
                sign *= -1;
            }
        }
        sign
    }
}

/* */

impl Solution {
    pub fn array_sign(nums: Vec<i32>) -> i32 {
        let mut count = 0;
        for &n in nums.iter() {
            if n == 0 {return 0;}
            if n < 0 {
                count += 1;
            }
        }
        if count & 1 == 0 {1} else {-1}
    }
}


